src/main/kotlin/com/lykke/me/subscriber/rabbitmq/MeRabbitMqProtoEventListener.kt

```diff
@@ -0,0 +1,106 @@
+package com.lykke.me.subscriber.rabbitmq
+
+import com.lykke.me.subscriber.AbstractMeListener
+import com.lykke.me.subscriber.config.RabbitMqConfig
+import com.lykke.me.subscriber.incoming.events.proto.MeProtoEvent
+import com.lykke.me.subscriber.incoming.events.proto.deserialization.MeProtoEventDeserializer
+import com.lykke.utils.logging.ThrottlingLogger
+import com.lykke.utils.rabbit.Connector
+import com.lykke.utils.rabbit.ConsumerFactory
+import com.lykke.utils.rabbit.RabbitMqSubscriber
+import com.rabbitmq.client.Channel
+import com.rabbitmq.client.ConnectionFactory
+import com.rabbitmq.client.Consumer
+import java.util.concurrent.LinkedBlockingQueue
+import kotlin.concurrent.thread
+import com.lykke.utils.rabbit.RabbitMqConfig as UtilsRabbitMqConfig
+
+class MeRabbitMqProtoEventListener(private val configs: Set<RabbitMqConfig>) : AbstractMeListener<MeProtoEvent<*>>() {
+
+    companion object {
+        private val LOGGER = ThrottlingLogger.getLogger(MeRabbitMqProtoEventListener::class.java.name)
+    }
+
+    private val queue = LinkedBlockingQueue<RmqMessageWrapper>()
```

Comment:
it would be greate if it is possible to monitor size of this queue. I suggest to provide api - user of this lib provides factory to create this queue. In such a way user will have access to this queue and monitor its size.

Reply:
fixed